internal/stacks/stackruntime/internal/stackeval/stubs/unknown.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package stubs
+
+import (
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/hashicorp/terraform/internal/providers"
+	"github.com/hashicorp/terraform/internal/tfdiags"
+)
+
+var _ providers.Interface = (*unknownProvider)(nil)
+
+// unknownProvider is a stub provider that represents a provider that is
+// unknown to the current Terraform configuration. This is used when a reference
+// to a provider is unknown, or the provider itself has unknown instances.
+//
+// This provider wraps an unconfigured provider client, which is used to handle
+// offline functionality.
+//
+// TODO: We can return more specific values than cty.DynamicVal.
```

Comment:
This is a todo for later I assume?

Reply:
Whoops, no I meant to do this now :/ Done!